galaxy_importer/collection.py

```diff
@@ -57,8 +58,20 @@ def _import_collection(file, filename, logger, cfg):
     with tempfile.TemporaryDirectory() as tmp_dir:
         sub_path = 'ansible_collections/placeholder_namespace/placeholder_name'
         extract_dir = os.path.join(tmp_dir, sub_path)
-        with tarfile.open(fileobj=file, mode='r') as pkg_tar:
-            pkg_tar.extractall(extract_dir)
+        os.makedirs(extract_dir)
+
+        cmd = [
+            'tar',
+            f'--directory={extract_dir}',
+            '-xf',
+            file.name,
+        ]
+        proc = Popen(cmd, cwd=PULP_ARTIFACT_DIR)
+        return_code = proc.wait()
+        if return_code != 0:
+            raise exc.ImporterError('A tar error occurred in {}, returncode={}'.format(
+                ' '.join(cmd), return_code))
```

Comment:
I would consider adding some error logging here with the return code / stdout / stderr / cmd.

That is logging to at least the "regular" app logger, and... maybe to the "import" logger that is used to collect and display import process.

Just for troubleshooting purposes and to be able to verify that any failures aren't exploit atttempts.

Reply:
(depending on what all the SubprocessError includes, just logging it's str/repr may be enough. I'd have to try it and see)

Reply:
Added more info in logging here, passing to display in import process